src/openms/include/OpenMS/CHEMISTRY/EnzymesDB.h

```diff
@@ -0,0 +1,178 @@
+
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2015.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Xiao Liang $
+// $Authors: Xiao Liang $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_CHEMISTRY_ENZYMESDB_H
+#define OPENMS_CHEMISTRY_ENZYMESDB_H
+
+#include <OpenMS/DATASTRUCTURES/Map.h>
+#include <boost/unordered_map.hpp>
+#include <OpenMS/DATASTRUCTURES/String.h>
+
+#include <set>
+
+namespace OpenMS
+{
+  // forward declarations
+  class Enzyme;
+
+  /** @ingroup Chemistry
+
+          @brief enzyme database which holds enzymes
+
+          The enzymes stored in this DB are defined in a
+          XML file under share/CHEMISTRY/Enzymes.xml
+
+  */
+  class OPENMS_DLLAPI EnzymesDB
+  {
+public:
+
+    /** @name Typedefs
+    */
+    //@{
+    typedef std::set<Enzyme *>::iterator EnzymeIterator;
+    typedef std::set<const Enzyme *>::const_iterator EnzymeConstIterator;
+    //@}
+
+    /// this member function serves as a replacement of the constructor
+    inline static EnzymesDB * getInstance()
+    {
+      static EnzymesDB * db_ = 0;
```

Comment:
sounds good, no need to put it all on @liangoaix to fix, this should be our issue